main/src/library/File.flix

```diff
@@ -840,4 +840,104 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Returns a list naming the files and directories in the directory `f`.
+    /// Only the names of the files and directories are specified.
```

Comment:
The reason I choose to have full path for `walk`, is because if we go into a sub-directory, it makes more sense to specify the full path to know how we got to a specific file, if that makes sense. If the file we find is at the bottom of the file-tree, it is nice to know the full path.

But for `list`, we only go through the first layer of directories and files. Only the direct children to the specified `f`. Thus we know where they are located unlike `walk`, and only need to know the names.

If we where to make it consistent for all functions, I would prefer to specify the full path, because that makes the most sense for `walk` in my opinion. And it should not be too confusing to have `list` specify the full path.

what do you think?

Reply:
That's a good point about `walk`.

For `list`, I think my instinct as a user is to want to do something like `File.list(myDir) |> List.map(File.read) |> ...`. So to do this cleanly I need `list` to return the entire path.

Reply:
You are right. For consistency and usability let's specify full path